src/Images.jsx

```diff
@@ -46,10 +47,10 @@ class Images extends React.Component {
         if (imageTag)
             pullImageId += ":" + imageTag;
 
-        this.setState({ imageDownloadInProgress: imageName });
+        this.setState((previous) => ({ imageDownloadInProgress: [...previous.imageDownloadInProgress, imageName] }));
```

Comment:
Style nitpick throughout the patch: Please avoid 1-tuples. `previous`, not `(previous)`.

Reply:
This is still current, note the "throughout the patch"